crates/rethnet_eth/src/remote/client.rs

```diff
@@ -124,14 +127,20 @@ impl RpcClient {
         }
     }
 
-    fn write_response_to_cache(&self, request_body: &str, response: &str) {
-        std::fs::write(self.make_cache_path(request_body), response.as_bytes())
-            .expect("failed to write to on-disk RPC response cache");
+    fn write_response_to_cache(&self, cache_key: Option<&CacheKey>, response: &str) {
+        if let Some(cache_key) = cache_key {
+            std::fs::write(self.make_cache_path(cache_key), response.as_bytes())
+                .expect("failed to write to on-disk RPC response cache");
+        }
     }
 
     /// returns response text
-    async fn send_request_body(&self, request_body: &str) -> Result<String, RpcClientError> {
-        if let Some(cached_response) = self.read_response_from_cache(request_body) {
+    async fn send_request_body(
+        &self,
+        request_body: &str,
+        cache_key: Option<CacheKey>,
+    ) -> Result<String, RpcClientError> {
+        if let Some(cached_response) = self.read_response_from_cache(cache_key.as_ref()) {
             Ok(cached_response)
```

Comment:
```suggestion
        if let Some(cache_key) = &cache_key {
            self.read_response_from_cache(cache_key)
```

Reply:
I would like to avoid making this change to minimize conflicts with #4270. It's fixed there.